packages/extension-api/src/extension-api.d.ts

```diff
@@ -4360,7 +4360,18 @@ declare module '@podman-desktop/api' {
 
   export type CliToolState = 'registered';
 
-  export interface CliTool extends Disposable {
+  export interface CliTool extends CliToolInfo, Disposable {
+    updateVersion(version: CliToolUpdateOptions): void;
+    onDidUpdateVersion: Event<string>;
+
+    // register cli update flow
+    registerUpdate(update: CliToolUpdate | CliToolSelectUpdate): Disposable;
+
+    // register cli installer
+    registerInstaller(installer: CliToolInstaller): Disposable;
+  }
+
+  export interface CliToolInfo {
```

Comment:
I am wondering if we should not include the version in some way ? If I need at some point to require a specific cli tool, I feel like I would check the version to ensure compatibility with my code

Reply:
I was looking at the code, and it feel odd that when registering a cli tool we need to provide a version (or undefined); but the version field is not in the Cli object.

Reply:
This could be a follow up PR as this is a problem that was already here before